solc/CommandLineParser.cpp

```diff
@@ -451,6 +455,13 @@ void CommandLineParser::parseOutputSelection()
 			CompilerOutputs::componentName(&CompilerOutputs::ewasm),
 			CompilerOutputs::componentName(&CompilerOutputs::ewasmIR),
 		};
+		static set<string> const evmAssemblyJsonImportModeOutputs = {
```

Comment:
@chriseth @ekpyron does the restriction of the output modes make sense for you too?

Reply:
@cameel can you answer this?

Reply:
Yeah, looks sensible. I think they all should be possible to generate from imported assembly.

My only doubt would be about `binaryRuntime`. If it comes from compiling a contract, we do have the construction/runtime code split. Do we always have it with arbitrary assembly though?

Reply:
> Also for [#12704 (comment)](https://github.com/ethereum/solidity/pull/12704#discussion_r826422552) - where my assumptions correct to limit and/or support these?

This second check is redundant. It's just the opposite of the supported input modes, isn't it?